src/mutable_data.rs

```diff
@@ -740,6 +740,18 @@ impl Data {
             Data::Unseq(data) => data.owner,
         }
     }
+
+    pub fn mutate_entries(&mut self, actions: EntryActions, requester: PublicKey) -> Result<()> {
+        match (self, actions) {
+            (Data::Seq(data), EntryActions::Seq(ref actions)) => {
+                data.mutate_entries(actions.clone(), requester)
```

Comment:
`actions` is passed by value so I don't think the `clone` is necessary.

Reply:
If I try removing the `clone` I get this error:

```
error[E0507]: cannot move out of borrowed content
   --> src/mutable_data.rs:747:37
    |
747 |                 data.mutate_entries(*actions, requester)
    |                                     ^^^^^^^^ cannot move out of borrowed content
```

If I try not taking `actions` by reference I get this:

```
error[E0009]: cannot bind by-move and by-ref in the same pattern
   --> src/mutable_data.rs:749:53
    |
749 |             (Data::Unseq(data), EntryActions::Unseq(actions)) => {
    |                          ----                       ^^^^^^^ by-move pattern here
    |                          |
    |                          both by-ref and by-move used
```

Don't see a way around these errors, so I left the code as-is and added a comment.

Reply:
Hm, just tried in in rust playground. Apparently it's not possible to match by reference and by move in the same pattern. A way around this could be to restructure it somehow like this:

```rust
match self {
    Data::Seq(data) => if let EntryActions::Seq(actions) = actions {
        return data.mutate_entries(actions, requester);
    },
    Data::Unseq(data) => if let EntryAction::Unseq(actions) = actions {
        return data.mutate_entries(actions, requester);
    }
}

Error(Err::InvalidOperation)
```